packages/fiber/src/core/index.tsx

```diff
@@ -335,34 +335,42 @@ function Portal({
    *  <Canvas>
    *    {createPortal(...)} */
 
-  const { events, ...rest } = state
   const previousRoot = useStore()
   const [raycaster] = React.useState(() => new THREE.Raycaster())
   const [pointer] = React.useState(() => new THREE.Vector2())
 
   const inject = React.useCallback(
-    (state: RootState, injectState?: RootState) => {
-      const intersect: Partial<RootState> = { ...state }
+    (prevRootState: RootState, injectState?: RootState) => {
+      const { events, ...rest } = state
+      const intersect: Partial<RootState> = { ...prevRootState }
 
       if (injectState) {
         // Only the fields of "state" that do not differ from injectState
-        Object.keys(state).forEach((key) => {
+
+        const protectedKeys: Readonly<Record<string, string>> = {
+          size: 'size',
+          viewport: 'viewport',
+          internal: 'internal',
+          performance: 'performance',
+        }
+
+        for (let key in prevRootState) {
```

Comment:
Bring back the protected keys as requested. It's now an object (it was an array) and also moved to outside the loop